src/dioptra/client/sessions.py

```diff
@@ -118,7 +118,9 @@ def convert_response_to_dict(response: DioptraResponseProtocol) -> dict[str, Any
             response.request.url,
             response.text,
         )
-        raise StatusCodeError(f"Error code returned: {response.status_code}")
+        raise StatusCodeError(
+            f"Error code returned: {response.status_code}", response.text
```

Comment:
Recommend that we prepend a statement to contextualize what `response.text` should be understood as:

```suggestion
            f"Error code returned: {response.status_code}",
            f"Error message returned: {response.text}",
```